src/audio/drc/Testbench_drc_lin2dB/drc_lin2dB_fixed.c

```diff
@@ -0,0 +1,137 @@
+
+
+
+int main()
+{
+	int32_t l_testvector, x[TEST_VECTOR], y[TEST_VECTOR]; 	/*Increase/decrease length as per need*/
+    uint16_t idx = 0;
+	int i;
+	/*int mkdir(const char* dirname);*/
+	mkdir("Results", 0777);
+	FILE* fd = fopen("Results/file.txt", "w");
+	/*fprintf(fd, " %10s| %10s|%10s|\n-----------+-----------+--------------+\n",  "idx", "testvector", "Fixlog10linear");*/
+	fprintf(fd, " %10s  %10s %10s \n", "idx", "testvector", "Fixlog10linear");
+
+    data_initialization_fixpt(x);
+    input_data_fixpt(x, y);
+	for (i = 0; i < TEST_VECTOR; i++)
+	{
+		l_testvector = y[i];
+		/* Data logging purpose_start*/
+        idx++;
+		fprintf(fd, " %10d %11li ", idx, l_testvector);
+		drc_lin2db_fixed(l_testvector, idx,fd);
+		/* Data logging purpose*/
+	}
+	fclose(fd);
+}
+
+/*
+ * function[y]= input_data_fixpt(x)
+ * Arguments    : const int32_t x[TEST_VECTOR]
+ *                int32_t y[TEST_VECTOR]
+ * Return Type  : void
+ */
+void input_data_fixpt(const int32_t x[TEST_VECTOR], int32_t y[TEST_VECTOR])
+{
+    memcpy(&y[0], &x[0], 639U * (sizeof(int32_t)));
+}
+
+/*
+ * Input is Q6.26: max 32.0
+ * Output range ~ (-inf, 30.1030); regulated to Q11.21: (-1024.0, 1024.0)
+ */
+
+inline int32_t drc_lin2db_fixed(int32_t linear, uint16_t idx, FILE* fd)		/* Data logging purpose*/
+{
+	int32_t log10_linear;
+	/* For negative or zero, just return a very small dB value. */
+	if (linear <= 0){
+		fprintf(fd, " %13.8f \n", -0.00047684);
+		return Q_CONVERT_FLOAT(-1000.0f, 21);
+	}
+	else{
+		log10_linear = log10_fixed(linear); /* Q6.26 */
+		log10_linear_log[idx] = log10_linear;		/* Dbg buffer*/
+		fprintf(fd, " %13li\n", log10_linear);
+		return q_mult(20, log10_linear, 0, 26, 21);	/*Q11.21*/ /*-67108866/2^21 = 32.0dB*/
+	}
+}
+/*
+ * Input depends on precision_x
+ * Output range [0.5, 1); regulated to Q2.30
+ */
+static inline int32_t rexp_fixed(int32_t x, int precision_x, int* e)
+{
+	int bit = 31 - norm_int32(x);
+
+	*e = bit - precision_x;
+
+	if (bit > 30)
+		return Q_SHIFT_RND(x, bit, 30);
+	if (bit < 30)
+		return Q_SHIFT_LEFT(x, bit, 30);
+	return x;
+}
+/*
+ * Input is Q6.26: max 32.0
+ * Output range ~ (-inf, 1.505); regulated to Q6.26: (-32.0, 32.0)
+ */
+static inline int32_t log10_fixed(int32_t x)
+{
+#define qc 26
+	/* Coefficients obtained from:
+	 * fpminimax(log10(x), 5, [|SG...|], [1/2;sqrt(2)/2], absolute);
+	 * max err ~= 6.088e-8
+	 */
+	const int32_t ONE_OVER_SQRT2 = Q_CONVERT_FLOAT(0.70710678118654752f, 30); /* 1/sqrt(2) */
+	const int32_t A5 = Q_CONVERT_FLOAT(1.131880283355712890625f, qc);
+	const int32_t A4 = Q_CONVERT_FLOAT(-4.258677959442138671875f, qc);
+	const int32_t A3 = Q_CONVERT_FLOAT(6.81631565093994140625f, qc);
+	const int32_t A2 = Q_CONVERT_FLOAT(-6.1185703277587890625f, qc);
+	const int32_t A1 = Q_CONVERT_FLOAT(3.6505267620086669921875f, qc);
+	const int32_t A0 = Q_CONVERT_FLOAT(-1.217894077301025390625f, qc);
+	const int32_t LOG10_2 = Q_CONVERT_FLOAT(0.301029995663981195214f, qc);
+	int e;
```

Comment:
any reason why e is an int and not int32_t ?

Reply:
I will change int to int32_t